### ISSUE TITLE
can you tell me the mean of decode_head and auxiliary_head?

### ISSUE THREAD

**Issue by zhujiesuper** [open-mmlab/mmsegmentation#53]

**Comment by xvjiarui** [comment]
Hi @zhujiesuper 
`decode_head` is for decoding semantic segmentation output. 
`auxiliary_head` is just adding an auxiliary loss.

**Comment by zhujiesuper** [comment]
thank you! but if all the model need auxiliary_head? i find it in all model,but in papers do not all have ,i know pspnet have an auxiliary loss,but others do not have,can you tell me how to handle this

**Comment by xvjiarui** [comment]
Hi @zhujiesuper 
For fair comparison, we use auxiliary loss for all methods. 
Note that, there are many implementation details in our repo that differ from the official repo.

**Comment by baibaidj** [comment]
> thank you! but if all the model need auxiliary_head? i find it in all model,but in papers do not all have ,i know pspnet have an auxiliary loss,but others do not have,can you tell me how to handle this

The auxilliary_head is modular and optional, which means that you can remove it if you don't want it. 
 `model = dict(
    type='EncoderDecoder', # tells registery to use which module to build this model
    pretrained='open-mmlab://resnet50_v1c',
    backbone=dict(
        type='ResNetV1c',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        dilations=(1, 1, 2, 4),
        strides=(1, 2, 1, 1),
        norm_cfg=norm_cfg,
        norm_eval=False,
        style='pytorch',
        contract_dilation=True),
    decode_head=dict(
        type='PSPHead', # type can be seen as the surrogates for python classes to call from
        in_channels=2048,
        in_index=3,
        channels=512,
        pool_scales=(1, 2, 3, 6),
        dropout_ratio=0.1,
        num_classes=19,
        norm_cfg=norm_cfg,
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    auxiliary_head=dict(
        type='FCNHead',
        in_channels=1024,
        in_index=2,
        channels=256,
        num_convs=1,
        concat_input=False,
        dropout_ratio=0.1,
        num_classes=19,
        norm_cfg=norm_cfg,
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4)))
# model training and testing settings
train_cfg = dict()
test_cfg = dict(mode='whole')`

For the config of PSPNet as shown above, you just need to remove all the content of `auxiliary_head` variable. Then the model won't compute the auxiliary loss any more.

**Comment by zhujiesuper** [comment]
does it mean i delete 

auxiliary_head=dict(
type='FCNHead',
in_channels=1024,
in_index=2,
channels=256,
num_convs=1,
concat_input=False,
dropout_ratio=0.1,
num_classes=19,
norm_cfg=norm_cfg,
align_corners=False,
loss_decode=dict(
type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4))
)

and it work?

**Comment by xvjiarui** [comment]
Hi @zhujiesuper 
Yes. Deleting `auxilary_head` in cfg will remove auxiliary head in the model.
